src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -848,19 +863,29 @@ protected DynamoModel(IStartConfiguration config)
             pathManager.Preferences = PreferenceSettings;
             PreferenceSettings.RequestUserDataFolder += pathManager.GetUserDataFolder;
 
-            SearchModel = new NodeSearchModel(Logger);
-            SearchModel.ItemProduced +=
-                node => ExecuteCommand(new CreateNodeCommand(node, 0, 0, true, true));
+            if (!IsServiceMode)
+            {
+                SearchModel = new NodeSearchModel(Logger);
+                SearchModel.ItemProduced +=
+                    node => ExecuteCommand(new CreateNodeCommand(node, 0, 0, true, true));
+            }
 
             NodeFactory = new NodeFactory();
             NodeFactory.MessageLogged += LogMessage;
 
             //Initialize the ExtensionManager with the CommonDataDirectory so that extensions found here are checked first for dll's with signed certificates
             extensionManager = new ExtensionManager(new[] { PathManager.CommonDataDirectory });
             extensionManager.MessageLogged += LogMessage;
-            var extensions = config.Extensions ?? LoadExtensions();
+            var extensions = config.Extensions ?? new List<IExtension>();
+            if (!extensions.Any() && !IsServiceMode)
```

Comment:
Is there a list somewhere with all our default extensions and what are they doing ?

Reply:
In code, we dont maintain a list. We only maintain the list using xml manifest files in `extensions` and `viewextensions` folder